File: Challenges/5kyu/rgb_to_hex_conversion/typescript/solution.ts
export function rgb(r: number, g: number, b: number): string {
  let hex = "";
  
  hex += byteToHex(r);
  hex += byteToHex(g);
  hex += byteToHex(b);
  
  return hex;
}

function byteToHex(byte:number):string {
  let digits = "0123456789ABCDEF";
  let hex = "";
  
  byte = (byte < 0) ? 0 : (byte > 255) ? byte = 255 : byte;
  hex += digits[(byte - (byte % 16)) / 16];
  hex += digits[byte % 16];
  
  return hex;
}

File: Challenges/5kyu/rgb_to_hex_conversion/python/solution.py
def rgb(r, g, b):
    hex = ""
    
    hex += byteToHex(r)
    hex += byteToHex(g)
    hex += byteToHex(b)

    return hex
    pass

def byteToHex(byte):
  digits = "0123456789ABCDEF"
  hex = ""
  
  byte = 0 if byte < 0 else 255 if byte > 255 else byte
  hex += digits[int((byte - (byte % 16)) / 16)]
  hex += digits[int(byte % 16)]
  
  return hex